dialog/base.go

```diff
@@ -228,6 +221,45 @@ func newButtonList(buttons ...*widget.Button) fyne.CanvasObject {
 	return list
 }
 
+// ===============================================================
+// ThemedBackground
+// ===============================================================
+
+type themedBackground struct {
+	widget.BaseWidget
+}
+
+func newThemedBackground() *themedBackground {
+	t := &themedBackground{}
+	t.ExtendBaseWidget(t)
+	return t
+}
+
+func (t *themedBackground) CreateRenderer() fyne.WidgetRenderer {
+	t.ExtendBaseWidget(t)
+	rect := canvas.NewRectangle(theme.BackgroundColor())
+	return &themedBackgroundRenderer{rect, []fyne.CanvasObject{rect}}
+}
+
+type themedBackgroundRenderer struct {
+	rect    *canvas.Rectangle
+	objects []fyne.CanvasObject
+}
+
+func (renderer *themedBackgroundRenderer) Objects() []fyne.CanvasObject { return renderer.objects }
+func (renderer *themedBackgroundRenderer) Destroy()                     {}
+func (renderer *themedBackgroundRenderer) Layout(size fyne.Size)        { renderer.rect.Resize(size) }
+func (renderer *themedBackgroundRenderer) MinSize() fyne.Size           { return renderer.rect.MinSize() }
+func (renderer *themedBackgroundRenderer) Refresh() {
```

Comment:
AFAIK we do not have one-liner methods up to now. I’m not happy with those, probably we should come to a core team decision and add it to the style guidelines, @andydotxyz?

Also, the methods should be ordered alphabetically.

Reply:
> AFAIK we do not have one-liner methods up to now. I’m not happy with those

I see, not a problem, I can update them :). I used that style to avoid using too much space, but maybe the other form is more clear.

> Also, the methods should be ordered alphabetically.

Got it. I forgot to sort them.

Reply:
I added:

```
* Functions and other scope blocks (if, for etc) should not be arranged on one line - a scope body belongs on a new line.
```

Reply:
done